vendor/libtuntap-master/tuntap-unix.c

```diff
@@ -169,7 +169,7 @@ tuntap_set_ifname(struct device *dev, const char *ifname)
   }
 
   (void)memset(dev->if_name, 0, IF_NAMESIZE);
-  (void)strncpy(dev->if_name, ifname, len);
+  (void)strncpy(dev->if_name, ifname, len + 1);
```

Comment:
should be `sizeof(dev->if_name)` right?

Reply:
rather `min(sizeof(dev->ifname), len)`

Reply:
that's wrong those, because if `ifname` is as long as `IF_NAMESIZE`, there is no room for the trailing '\0'

Reply:
`len` is calculated based of `ifname` and it it exceeds that it needs to truncate

Reply:
`ifname` not to be confused with `dev->if_name`

Reply:
So consider this case:

1. `IF_NAMESIZE` is 4 (just to make this easier for me to type out)
2. `ifname` is `loki\0` (making `strlen(ifname)` == 4)


```
char arr[IF_NAMESACE];
strncpy(arr, ifname, strlen(ifname));
```

at the end of this, `arr` is `['l', 'o', 'k', 'i']`, without any trailing '\0' character (making it an invalid c-string)

Reply:
`len` is calculated based on `ifname`  (user input) not `dev->if_name` (target buffer)